megalinter/MegaLinter.py

```diff
@@ -428,15 +428,9 @@ def list_files_all(self):
         ]
         if logging.getLogger().isEnabledFor(logging.DEBUG):
             logging.debug("Root dir content:\n" + "\n- ".join(all_files))
-        excluded_directories = utils.list_excluded_directories()
-        for (dirpath, _dirnames, filenames) in os.walk(self.workspace):
-            exclude = False
-            for dir1 in dirpath.split(os.path.sep):
-                if dir1 in excluded_directories:
-                    exclude = True
-                    break
-            if exclude is True:
-                continue
+        excluded_directories = set(config.get_list("EXCLUDED_DIRECTORIES", utils.list_excluded_directories()))
```

Comment:
@kolomenkin could you do something like below, so it's an apprend and not a replace ?

```python
set(config.get_list("EXCLUDED_DIRECTORIES",[]) + utils.list_excluded_directories())
```

Reply:
Originally I wanted to change `list_excluded_directories`. The only one thing stopped me. It did not depend on `config` before. So I decided not to add this dependency. But this is a very minor argument in this case.

I will update `list_excluded_directories`. It will encapsulate calculation of this list better. And it will return a `set` from now, not a `list`